homeassistant/components/siren/__init__.py

```diff
@@ -51,14 +51,24 @@ class SirenTurnOnServiceParameters(TypedDict, total=False):
     volume_level: float
 
 
-def filter_turn_on_params(
+def process_turn_on_params(
     siren: SirenEntity, params: SirenTurnOnServiceParameters
 ) -> SirenTurnOnServiceParameters:
-    """Filter out params not supported by the siren."""
+    """
+    Process turn_on service params.
+
+    Filters out unsupported params and validates the rest.
+    """
     supported_features = siren.supported_features or 0
 
     if not supported_features & SUPPORT_TONES:
         params.pop(ATTR_TONE, None)
+    elif ATTR_TONE in params and (
+        not siren.available_tones
+        or (tone := params[ATTR_TONE]) not in siren.available_tones
+    ):
+        raise ValueError(f"Tone {tone} is not a valid tone for this device")
```

Comment:
I'd write:
```py
raise ValueError(f"Invalid tone received: {tone}")
```

I don't think we should include the word device in the context of entities. That makes it confusing with our other Home Assistant terminology.